src/ledger/LedgerTxn.cpp

```diff
@@ -2089,6 +2089,12 @@ void
 LedgerTxn::Impl::updateEntryIfRecorded(InternalLedgerKey const& key,
                                        bool effectiveActive)
 {
+    if (key.type() != InternalLedgerEntryType::LEDGER_ENTRY ||
```

Comment:
This looks right to me. But you should add a big comment here explaining that it's purely an optimization (this will help future maintainers know they can remove it if `updateEntry` changes) and why it's right.

Reply:
Done